errors_test.go

```diff
@@ -485,3 +485,94 @@ func TestGetDuckDBErrorIs(t *testing.T) {
 	require.NotErrorIs(t, invalidInputErr, outOfRangeErr1)
 	require.NotErrorIs(t, errors.New(errMsg), outOfRangeErr1)
 }
+
+func TestUnionErrors(t *testing.T) {
+	t.Parallel()
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	t.Run("empty union type", func(t *testing.T) {
+		_, err := db.Query("SELECT (1)::UNION() as empty_union")
+		var de *Error
+		require.True(t, errors.As(err, &de))
+		require.Equal(t, ErrorTypeParser, de.Type)
+	})
+
+	t.Run("duplicate member names", func(t *testing.T) {
+		_, err := db.Query("SELECT (1)::UNION(val INTEGER, val VARCHAR) as dup_union")
+		var de *Error
+		require.True(t, errors.As(err, &de))
+		require.Equal(t, ErrorTypeParser, de.Type)
+	})
+
+	t.Run("type mismatch", func(t *testing.T) {
+		_, err := db.Query("SELECT (true)::UNION(num INTEGER, str VARCHAR) as mismatch_union")
+		var de *Error
+		require.True(t, errors.As(err, &de))
+		require.Equal(t, ErrorTypeConversion, de.Type)
+	})
+
+	t.Run("nested union type", func(t *testing.T) {
+		_, err := db.Query(`
+            WITH inner_union AS (
+                SELECT (1)::UNION(num INTEGER, str VARCHAR) as u
+            )
+            SELECT (u)::UNION(nested UNION(num INTEGER, str VARCHAR), other INTEGER)
+            FROM inner_union
+        `)
+		var de *Error
+		require.True(t, errors.As(err, &de))
+		require.Equal(t, ErrorTypeConversion, de.Type)
+	})
+
+	t.Run("union in table creation", func(t *testing.T) {
+		// Test that unions work in table creation
+		_, err := db.Exec(`
+            CREATE TABLE test_union (
+                u UNION(num INTEGER, str VARCHAR)
+            )
+        `)
+		require.NoError(t, err)
+
+		// Test inserting wrong type
+		_, err = db.Exec(`
+            INSERT INTO test_union VALUES ((true)::UNION(num INTEGER, str VARCHAR))
+        `)
+		var de *Error
+		require.True(t, errors.As(err, &de))
+		require.Equal(t, ErrorTypeConversion, de.Type)
+	})
```

Comment:
These are DuckDB errors and not necessarily related to go-duckdb's `UNION` implementation. I don't think we necessarily need them. Same for `"invalid member type"` and `"missing member name"`.

Reply:
Good point. Thanks.